frontend/src/components/viewers/MetricsVisualizations.tsx

```diff
@@ -130,6 +177,38 @@ function getVerifiedMetricsArtifacts(
   );
 }
 
+function getV1VisualizationArtifacts(
+  linkedArtifacts: LinkedArtifact[],
+  artifactTypes: ArtifactType[],
+): LinkedArtifact | undefined {
+  if (!linkedArtifacts || !artifactTypes) {
+    return undefined;
+  }
+  const systemArtifacts = filterLinkedArtifactsByType(
+    'system.Artifact',
+    artifactTypes,
+    linkedArtifacts,
+  );
+
+  const v1VisualizationArtifacts = systemArtifacts.filter(x => {
+    if (!x) {
+      return false;
+    }
+    const artifactName = getArtifactName(x);
+    // This is a hack to find mlpipeline-ui-metadata artifact for visualization.
+    const updatedName = artifactName?.replace(/_/g, '-').toLowerCase();
```

Comment:
For discussion, I think we can be more aggressive in sanitization.
e.g. convert "MLPipeline UI Metadata" also to `mlpipeline-ui-metadata`.

Basically, convert any non-alphanumeric character to `-`. This is consistent with KFP v1 compiler.

Reply:
I am going to adopt the aggressive sanitization as suggested. I think it will be great to track the changes in SDK in the future, so we can reflect changes in FE as well. adding to https://github.com/kubeflow/pipelines/issues/5830